app/controllers/dashboard_controller.rb

```diff
@@ -0,0 +1,7 @@
+class DashboardController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
We should add the before_action to ApplicationController. Requests that don't need it can use [skip_before_action](https://api.rubyonrails.org/classes/AbstractController/Callbacks/ClassMethods.html#method-i-skip_after_action)